chain/epoch_manager/src/lib.rs

```diff
@@ -410,6 +411,19 @@ impl EpochManager {
             )
         };
         let next_next_epoch_config = self.config.for_protocol_version(next_version);
+        #[cfg(feature = "sandbox")]
+        {
+            // explaination on calculation:
+            // (block_height / length) gives us epoch_height, but we index from 1, so +1 at the end.
+            // Note, this is a retroactive update for sandbox since we can receive fast-forward
+            // requests in the middle of an epoch. block_height is different based on when it gets
+            // called. When fast-forwarded, we need to retroactively adjust the epoch height, but
+            // with a normal `finalize_epoch`, it's the start of the epoch's next block height, so
+            // (height - 1) is needed after adding +1 at the end.
```

Comment:
do you have any doc describing how the chain looks like after the 'fast forward'?

Because I see that you change some things, but don't change the others 
(for example: here you'd update the epoch_height -- but this might also mean that we'll have 'holes' in the epoch history).

The approach, I'd have preferred is that we do move timestamp forward (fine) - but for blocks, what we do - is we generate 2-3 blocks per epoch (to get finality and generate the next list of validators) - and jump to another epoch.

This would mean that such chain is still a 'correct' chain.

Reply:
how the chain looks like after the fast forward (after like 10k blocks) is the following:
```
block_timestamp: T + 10000 * avg_block_prod_time
block_height: H + 10000
epoch_height: E + 10000 / epoch_length
```
where the variables before fast forwarding are the following:
```
block_timestamp: T
block_height: H
epoch_height: E
```

I'm not sure I follow with the holes. Do you mean not being able to refer to a previous epoch? The history would just end up looking like the following with epoch length of 5:
```
E1 ........... E2 ..... E11 .............
B1 B2 B3 B4 B5 B6 B7 B8 B50 B51 B52 ...
                        ^^^
                        fast forward request w/ 42 blocks
```
where
```
E = epoch height
B = block height
```

We'd still be able to refer to the previous epoch, since this is all being referenced via the epoch-id, which hasn't been changed. So I'm not sure what a "correct" chain entails wrt epochs/blocks. What would one look like, or any docs I can look at?